### DISCUSSION TITLE
Is there any good API for collision detection between two objects in isaac lab?

### DISCUSSION THREAD

**Discussion by shaoxiang** [isaac-sim/IsaacLab#560] [Q&A]
Contact sensors or ray caster sensors can perform collision detection, but this requires adding sensors to objects. Is there a simple way to perform collision detection between two objects?
I used the get_physx_scene_query_interface().overlap_shape method in isaac sim before, and I want to know if there is a better physx collision detection API in isaac lab.

**Comment by savevska** [comment]
Hi @shaoxiang,

I was wondering if you were able to find a collision detection API for Isaac Lab? The overlap_mesh method doesn’t detect self-collisions in Isaac Lab, unfortunately.

Any assistance would be greatly appreciated.

**Comment by Mayankm96** [comment]
There isn't a mechanism in the isaac lab to detect shape collisions (beyond what already exists in Omniverse). You can make your own warp kernels or operations for fast and exclusive collision checks probably.

Additionally, I don't think the `get_physx_scene_query_interface()` works when you have Fabric enabled. If you disable it, you can use it the same way as how you did it in Isaac Sim.

**Reply by savevska** [reply]
Thank you for your response, @Mayankm96. 
I apologize for bringing this up again, but I'm still struggling to find a solution.

The problem I'm facing is that I cannot use the contact report callback API because the asset has overlapping links, and enabling self-collision doesn't work. 
On the other side, the overlap_mesh API yields different results than what I observe in Isaac Sim, i.e. it fails to detect self-collisions.
The simulation configuration includes `enable_scene_query_support=True, use_fabric=False`.

I would appreciate your help in understanding this issue.
